commit f7f1394576563b7096fe858fb178bcaedfa0b245
Author: David Howells <dhowells@redhat.com>
Date:   Mon May 20 21:51:43 2019 +0100

    keys: Grant Link permission to possessers of request_key auth keys
    
    Grant Link permission to the possessers of request_key authentication keys,
    thereby allowing a daemon that is servicing upcalls to arrange things such
    that only the necessary auth key is passed to the actual service program
    and not all the daemon's pending auth keys.
    
    Signed-off-by: David Howells <dhowells@redhat.com>
    Reviewed-by: James Morris <jamorris@linux.microsoft.com>

diff --git a/security/keys/request_key_auth.c b/security/keys/request_key_auth.c
index 572c7a60473a..ec5226557023 100644
--- a/security/keys/request_key_auth.c
+++ b/security/keys/request_key_auth.c
@@ -204,7 +204,7 @@ struct key *request_key_auth_new(struct key *target, const char *op,
 
 	authkey = key_alloc(&key_type_request_key_auth, desc,
 			    cred->fsuid, cred->fsgid, cred,
-			    KEY_POS_VIEW | KEY_POS_READ | KEY_POS_SEARCH |
+			    KEY_POS_VIEW | KEY_POS_READ | KEY_POS_SEARCH | KEY_POS_LINK |
 			    KEY_USR_VIEW, KEY_ALLOC_NOT_IN_QUOTA, NULL);
 	if (IS_ERR(authkey)) {
 		ret = PTR_ERR(authkey);